src/System.Collections.Specialized/src/System/Collections/Specialized/OrderedDictionary.cs

```diff
@@ -24,12 +24,12 @@ namespace System.Collections.Specialized
     [System.Runtime.CompilerServices.TypeForwardedFrom("System, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089")]
     public class OrderedDictionary : IOrderedDictionary, ISerializable, IDeserializationCallback
     {
-        private ArrayList _objectsArray;
-        private Hashtable _objectsTable;
+        private ArrayList _objectsArray = null!;
+        private Hashtable _objectsTable = null!;
```

Comment:
These should be `?` rather than being initialized to `null!`.

Reply:
It's a change to the source, but if you want to make EnsureObjectsArray return `ArrayList` and then use that in the call sites, that'd be ok, e.g. EnsureObjectsArray would change to:
```C#
private void EnsureObjectsArray() => _objectsArray ??= new ArrayList();
```
and then for example IndexOfKey would look like:
```C#
private int IndexOfKey(object key)
{
    if (_objectsArray == null)
    {
        return -1;
    }

    ArrayList objectsArray = EnsureObjectsArray();
    ... // all subsequent accesses use objectsArray rather than _objectsArray
}
```

Reply:
Updated all including _objectsTable, please re review